### DISCUSSION TITLE
Trying to set up Javascript transformer

### DISCUSSION THREAD

**Discussion by kgarland9363** [nextgenhealthcare/connect#4833] [Q&A] [answered]
I know this is probably an easy thing to figure out but I am trying to set up a transformer script that will look at the contents within the ZEI.3.1 segment and if the contents equal housewife, retired, disabled, etc. I want to leave the message alone but if the contents of ZEI.3.1 equal a company name, I want to transform ZEI.2.1 to convent to "EM" (which stands for employed).  I have played with different scripts but nothing is working. 

I had small success with this script but it changed the ZEI.3.1 segment in all my messages - not just the ones containing a company name.

if (msg[“ZEI”][“ZEI.2”][“ZEI.2.1”].toString() != “amd”, “996”, “954”, “990”, “998”, “mil”, “rm”, “rta”, “RTM”, “UAF”, “47U”, “UAA”, “UAR”, “CG9”, “USM”, “NGU”, “USN”, “99U”, “995”, “991”, “997”) {
msg['ZEI']['ZEI.2']['ZEI.2.1'] = "EM";
} else {
msg['ZEI']['ZEI.2']['ZEI.2.1'] = "";
}

ZEI|1|396|WAYNE HEALTHCARE|835 SWEITZER ST^^GREENVILLE^OH^45331|
ZRV|Y^|Y^20211028000000|^||77067|N

Any help would be greatly appreciated.

Thank you,
Karen Garland

**Comment by pacmano1** [comment, marked as answer]
```JS
var list_to_ignore = new java.util.ArrayList(['housewife','retired','disabled'])

if (list_to_ignore.contains(msg['ZEI']['ZEI.3']['ZEI.3.1'].toString())) {
	//ignore
} else {
    msg['ZEI']['ZEI.2']['ZEI.2.1'] = 'EM';
}
```
https://forums.mirthproject.io/forum/mirth-connect/support/12427-filter-with-multiple-values?p=78645#post78645 effectively describes something simiar.

The code I posted  assume ZEI does not repeat

**Reply by kgarland9363** [reply]
Thank you so much. I kept getting an illegal character error initially but I rekeyed the script and I no longer received the error message and my conversion is working properly. Thank you such much for your knowledgeable input.

**Reply by pacmano1** [reply]
I don't know if you fixed your code or used what I posted.    But a couple of things.  For your next post, please use code highlighting.  See https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

Also, the code you posted wipes out ```msg['ZEI']['ZEI.2']['ZEI.2.1']``` which isn't what you stated in your intial quesiton.

**Comment by kgarland9363** [comment]
This is actually what I used in my transformer.

var ignore = new java.util.ArrayList(["amd","996","954","990","998","mil","rm","rta","RTM","UAF","47U","UAA","UAR","CG9","USM","NGU","USN","99U","995","991","997"])
if (ignore.contains(msg["ZEI"]["ZEI.2"]["ZEI.2.1"].toString())) {//ignore
}	else{msg["ZEI"]["ZEI.2"]["ZEI.2.1"]= "EM"; 
}

It does work; however, it keeps creating a ZEI segment when it doesn't initially exist. I am now trying to work through that issue. I originally thought it would be best to look to the value in ZEI.3.1 but it was just as easy to look at the value contained in ZEI.2.1 to perform my conversion.

**Reply by pacmano1** [reply]
Again, please forrmat your code.  It should look like this.    I took your code and added three back ticks and a 'JS' to the beginning and three back ticks to the end.

```JS
var ignore = new java.util.ArrayList(["amd", "996", "954", "990", "998", "mil", "rm", "rta", "RTM", "UAF", "47U", "UAA", "UAR", "CG9", "USM", "NGU", "USN", "99U", "995", "991", "997"])
if (ignore.contains(msg["ZEI"]["ZEI.2"]["ZEI.2.1"].toString())) { //ignore
} else {
    msg["ZEI"]["ZEI.2"]["ZEI.2.1"] = "EM";
}
```
You can just test for the existance of the ZEI segment and either bypass the entire if..then above or nest the test for segment existance in the else.

Screenshot of the raw code used to format your code:

![image](https://user-images.githubusercontent.com/44065187/140985759-1106a16d-6b2c-49b6-8ed2-121cb5e3656d.png)
